server/test/article.spec.js

```diff
@@ -173,156 +173,173 @@ describe('Articles routes', () => {
+    });
+    it('should check if description not sent', async () => {
+      const res = await chai.request(app)
+        .put(`/api/v1/article/${slug}`)
+        .set({ Authorization: `Bearer ${userToken}` })
+        .send({
+          title: 'This is a title from test',
+          body: 'This is from test',
+        });
+      const { message } = res.body;
+      expect(res).to.have.status(200);
+      expect(message).to.be.equal('Article edited successfully');
+    });
+    it('should return 422 when title is empty', async () => {
+      const res = await chai.request(app)
+        .put(`/api/v1/article/${slug}`)
+        .set({ Authorization: `Bearer ${userToken}` })
+        .send({
+          description: 'This is a description from test',
+          title: '  ',
+          body: 'This is from test',
+        });
+      const { message } = res.body;
+      expect(res).to.have.status(422);
+      expect(message.title).to.be.equal('Title should not be empty');
+    });
+    it('should check if title not sent', async () => {
+      const res = await chai.request(app)
+        .put(`/api/v1/article/${slug}`)
+        .set({ Authorization: `Bearer ${userToken}` })
+        .send({
+          description: 'This is a description from test',
+          body: 'This is from test',
+        });
+      const { message } = res.body;
+      expect(res).to.have.status(200);
+      expect(message).to.be.equal('Article edited successfully');
+    });
+    it('should return 422 when body is empty', async () => {
+      const res = await chai.request(app)
+        .put(`/api/v1/article/${slug}`)
+        .set({ Authorization: `Bearer ${userToken}` })
+        .send({
+          description: 'This is a description from test',
+          title: 'This is a title from test',
+          body: ' ',
+        });
+      const { message } = res.body;
+      expect(res).to.have.status(422);
+      expect(message.body).to.be.equal('body should not be empty');
+    });
+    it('should check if body not sent', async () => {
+      const res = await chai.request(app)
+        .put(`/api/v1/article/${slug}`)
+        .set({ Authorization: `Bearer ${userToken}` })
+        .send({
+          description: 'This is a description from test',
+          title: 'This is a title from test',
+        });
+      const { message } = res.body;
+      expect(res).to.have.status(200);
+      expect(message).to.be.equal('Article edited successfully');
+    });
+    it('should return 400 when a user is not author an article ', async () => {
+      const res = await chai.request(app)
+        .put(`/api/v1/article/${slug}`)
+        .set({ Authorization: `Bearer ${alternateToken}` })
+        .send({
+          description: 'This is a description from test',
+          title: 'This is a title from test',
+          body: 'This is from test',
+        });
+      const { message } = res.body;
+      expect(res).to.have.status(400);
```

Comment:
Hi @BukkyOmo good job so far. I think unauthorized should be 401, not 400. What do you think?

Reply:
Okay, will implement that now.